### PR TITLE
add SpoonModelTree.expandAll

### PR DESCRIPTION
**Pull request by argius**
Hello, this is my first PR in spoon.

I implement a new feature "Expand all" to GUI.
The feature, in SpoonModelTree, expands all child nodes of a selected node recursively.

Thanks for your review.

Regards.

### CODE REVIEW

**Comment by GerardPaligot** [comment]
Hello @argius,

First, thanks for your first PR on Spoon! :)

I tested your PR and it works well. I'll make a code review later today but I already have two remarks:

- The depth of your method is really big. It is maybe a good idea to reduce it.
- Your method returns always `null`.

**Comment by argius** [comment]
Hello Gérard,

Thanks for your review and pointing out some problems in my code.
I will consider those points.

**Comment by argius** [comment]
I revised `expandAll` method as follows:

- The method returns the top of the expanded node. If not expanded, it returns `null`.
- Removed some unnecessary codes; cast of `DefaultMutableTreeNode`, catch block.
- Uses `SwingUtilities.invokeLater` instead `invokeAndWait`.

Thanks.

File: src/main/java/spoon/support/gui/SpoonModelTree.java [hunk 1]
```diff
@@ -284,6 +305,46 @@ public DefaultMutableTreeNode search() {
...
+			return null;
+		}
+		final ExecutorService executor = Executors.newSingleThreadExecutor();
+		executor.execute(new Runnable() {
+			public void run() {
+				try {
+					Queue<DefaultMutableTreeNode> q = new LinkedList<DefaultMutableTreeNode>();
+					q.add(node);
+					while (!q.isEmpty()) {
+						final DefaultMutableTreeNode n = q.poll();
```
Comment (GerardPaligot) [inline src/main/java/spoon/support/gui/SpoonModelTree.java]: Creates a method `expand(node)` seems a good idea to reduce the depth of this method.

Comment (argius) [inline src/main/java/spoon/support/gui/SpoonModelTree.java]: I got it.

**Pull request merged**